external/fv3fit/fv3fit/pytorch/graph/mpg_unet.py

```diff
@@ -0,0 +1,331 @@
+import torch
+import torch.nn as nn
+import dataclasses
+from .graph_builder import build_dgl_graph_with_edge
+from dgl.nn.pytorch import NNConv
+from fv3fit.pytorch.activation import ActivationConfig
+
+
+@dataclasses.dataclass
+class MPGraphUNetConfig:
+    """
+    Attributes:
+        depth: depth of U-net architecture maximum
+        min_filters: mimumum number of hidden channels after first convolution
+        edge_hidden_features: mimumum number of edge hidden channels
+        num_step_message_passing : number of message passing steps.
+        aggregator: type of aggregator, one of "mean", "sum", or "max"
+        pooling_size: size of the pooling kernel
+        pooling_stride: pooling layer stride
+        activation: activation function
+    """
+
+    num_step_message_passing: int
+    edge_hidden_features: int
+    depth: int = 1
+    min_filters: int = 4
+    aggregator: str = "mean"
+    pooling_size: int = 2
+    pooling_stride: int = 2
+    activation: ActivationConfig = dataclasses.field(
+        default_factory=lambda: ActivationConfig()
+    )
+
+    def build(
+        self, in_channels: int, out_channels: int, nx: int,
+    ):
+        return MPGraphUNet(
+            self, in_channels=in_channels, out_channels=out_channels, nx=nx
+        )
+
+
+class MPNNGNN(nn.Module):
+    def __init__(
+        self,
+        node_in_channels,
+        node_hidden_channels,
+        edge_hidden_channels,
+        num_step_message_passing,
+        activation,
+        aggregator,
+        nx,
+    ):
+        """
+        MPNN is introduced in `Neural Message Passing for Quantum Chemistry
+        <https://arxiv.org/abs/1704.01212>`__.
+
+        Args:
+        ----------
+        node_in_channels : int
+            Size for the input node channels.
+        node_hidden_channels : int
+            Size for the hidden node representations.
+        edge_hidden_channels : int
+            Size for the hidden edge representations.
+        num_step_message_passing : int
+            Number of message passing steps.
+        activation: activation function
+        aggregator: aggregator type, one of 'sum', 'mean', 'max'
+        nx: number of horizontal grid points on each tile of the cubed sphere
```

Comment:
Please format this docstring to follow Google style. Please put the type hints in the `__init__` call signature, instead of in the docstring.